crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -235,13 +237,15 @@ where
     #[allow(clippy::ptr_arg)]
     fn append_matching_block_logs(
         &self,
-        _logs: &mut Vec<Log>,
-        _filter: &FilteredParams,
+        all_logs: &mut Vec<Log>,
+        filter: &FilteredParams,
         block: Block,
+        receipts: Vec<Receipt>
     ) {
-        let _block_log_index: u32 = 0;
-        let _block_hash = block.hash_slow();
+        let block_log_index: u32 = 0;
+        let block_hash = block.hash_slow();
 
+        // TODO: unify code with pubsub
         // loop over all transactions in the block
         for tx in block.body {
```

Comment:
this should prob re-use the logic from subscription that u just impled?

Reply:
done